pkg/reconciler/taskrun/config.go

```diff
@@ -0,0 +1,391 @@
+package taskrun
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	// Default allocation timeout for dynamic platforms in seconds (10 minutes)
+	defaultAllocationTimeout = 600
+)
+
+// parsePlatformList parses a comma-separated list of platforms, validates each one, and returns a slice of valid
+// platforms. Handles trailing commas.
+func parsePlatformList(platformList string, platformType string) ([]string, error) {
+	if platformList == "" {
+		return []string{}, nil
+	}
+
+	platforms := strings.Split(platformList, ",")
+	result := []string{}
+	for i, platform := range platforms {
+		platform = strings.TrimSpace(platform)
+		if platform == "" {
+			// Allow empty strings only at the end (from trailing commas)
+			if i == len(platforms)-1 {
+				continue
+			}
+			return nil, fmt.Errorf("invalid %s platform '': empty platform in middle of list", platformType)
+		}
+		if err := validatePlatformFormat(platform); err != nil {
+			return nil, fmt.Errorf("invalid %s platform '%s': %w", platformType, platform, err)
+		}
+		result = append(result, platform)
+	}
+	return result, nil
+}
+
+// ClusterHostConfig represents the complete host configuration from ConfigMap
+type ClusterHostConfig struct {
+	LocalPlatforms         []string
+	DynamicPlatforms       map[string]DynamicPlatformConfig
+	DynamicPoolPlatforms   map[string]DynamicPoolPlatformConfig
+	StaticPlatforms        map[string]StaticHostConfig
+	DefaultInstanceTag     string
+	AdditionalInstanceTags map[string]string
+}
+
+// DynamicPlatformConfig holds configuration for a single dynamic platform
+type DynamicPlatformConfig struct {
+	Type              string `mapstructure:"type"`
```

Comment:
do we need the `mapstructure` annotation? I don't see mapstructure used for parsing the configuration.